### PR TITLE
Fix fake payment instruction

### PR DESCRIPTION
**Pull request by whit537**
From https://github.com/gratipay/gratipay.com/pull/4157#issuecomment-265231655 and https://github.com/gratipay/gratipay.com/pull/4130#issuecomment-265232184 ...

### CODE REVIEW

**Comment by whit537** [comment]
Assuming this lands in `master` first, the whitespace trim will affect #4214, which also makes that move.

**Comment by whit537** [comment]
Will it fail here? :]

**Comment by whit537** [comment]
> fake.date_time_between(start_date="-30y", end_date="now", tzinfo=None)
# datetime(2005, 5, 23, 20, 47, 45)

http://faker.readthedocs.io/en/master/providers/faker.providers.date_time.html

**Comment by whit537** [comment]
Seemingly(?) fixed under https://github.com/gratipay/gratipay.com/pull/4130#issuecomment-265232184 by 3dda2a1e92330d9fff52361043c3af298dd92863. Not what I would have expected since "now" is supposedly the default.

If the presence of bug varies with the time of day, then rerunning the build on #4157 that doesn't have the fix #4130 has—we might expect that to _also_ pass.

**Comment by whit537** [comment]
[Build](https://travis-ci.org/gratipay/gratipay.com/builds/181712993) restarted.

**Comment by whit537** [comment]
Build passes! I don't think we're out of the woods yet.

**Comment by whit537** [comment]
We're [using](https://github.com/gratipay/gratipay.com/blob/2017/requirements.txt#L36) an [old](https://pypi.python.org/pypi/fake-factory/0.3.2) version of the library that was [tagged](https://github.com/joke2k/faker/commit/57197ee9713739b65b248583ab9441987b4ea5f4) in 2013, before https://github.com/joke2k/faker/issues/282 was [fixed](https://github.com/joke2k/faker/pull/307). It seems that an upgrade to [Faker](https://pypi.python.org/pypi/Faker) is in order! :-)

**Comment by whit537** [comment]
Harumph. Passing locally. Can I blame it on caching at Travis?

**Comment by mattbk** [comment]
Man, I just figured we had the right version of faker. Huh.

**Comment by whit537** [comment]
> I just figured we had the right version of faker. Huh.

We upgrade dependencies manually, to avoid the risk that newer versions break things we were depending on. We could certainly do a better job of upgrading dependencies on a schedule as opposed to haphazardly when stuff like this happens—the generalized case of https://github.com/gratipay/inside.gratipay.com/issues/776.

**Comment by whit537** [comment]
> Can I blame it on caching at Travis?

Apparently not.

**Comment by mattbk** [comment]
> E   AttributeError: 'module' object has no attribute 'Provider'

https://github.com/joke2k/faker/issues/332 is this issue, but we're installing fraom .tar.gz.

File: requirements.txt [hunk 1]
```diff
@@ -33,7 +33,10 @@
 ./vendor/contextlib2-0.5.4.tar.gz
 ./vendor/raven-5.32.0.tar.gz
 
-./vendor/fake-factory-0.3.2.tar.gz
+    ./vendor/six-1.8.0.tar.gz
```
Comment (whit537) [inline requirements.txt]: Yes, but it needs to be installed before Faker since Faker depends on it.

**Comment by whit537** [comment]
Heh ... COS fixed it in https://github.com/CenterForOpenScience/osf.io/commit/381c85189f3fa6d42570d8e6c0101049be4bd1ae by pinning to version 0.5.3. Does that version include the bugfix we need for our own presenting issue?

**Comment by whit537** [comment]
```
___________________ ERROR collecting tests/py/test_stats.py ____________________
tests/py/test_stats.py:8: in <module>
    from gratipay.utils import fake_data
gratipay/utils/fake_data.py:19: in <module>
    faker = Factory.create()
env/lib/python2.7/site-packages/faker/factory.py:39: in create
    prov_cls, lang_found = cls._get_provider_class(prov_name, locale)
env/lib/python2.7/site-packages/faker/factory.py:50: in _get_provider_class
    provider_class = cls._find_provider_class(provider, locale)
env/lib/python2.7/site-packages/faker/factory.py:98: in _find_provider_class
    return provider_module.Provider
E   AttributeError: 'module' object has no attribute 'Provider'
------------------------------- Captured stdout --------------------------------
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.address
FAKER: using localized provider module
FAKER: available locales: ['cs_CZ', 'de_DE', 'el_GR', 'en', 'en_AU', 'en_CA', 'en_GB', 'en_US', 'es', 'es_ES', 'es_MX', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hi_IN', 'hr_HR', 'it_IT', 'ja_JP', 'ko_KR', 'ne_NP', 'nl_NL', 'no_NO', 'pl_PL', 'pt_BR', 'pt_PT', 'sk_SK', 'sl_SI', 'sv_SE', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.address.en_US
FAKER: provider_module: <module 'faker.providers.address.en_US' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/address/en_US/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.barcode
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.barcode
FAKER: provider_module: <module 'faker.providers.barcode' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/barcode/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.color
FAKER: using localized provider module
FAKER: available locales: ['en_US']
FAKER: path: faker.providers.color.en_US
FAKER: provider_module: <module 'faker.providers.color.en_US' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/color/en_US/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.company
FAKER: using localized provider module
FAKER: available locales: ['bg_BG', 'cs_CZ', 'de_DE', 'en_US', 'es_MX', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hr_HR', 'it_IT', 'ja_JP', 'ko_KR', 'no_NO', 'pt_BR', 'pt_PT', 'sk_SK', 'sl_SI', 'sv_SE', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.company.en_US
FAKER: provider_module: <module 'faker.providers.company.en_US' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/company/en_US/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.credit_card
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.credit_card
FAKER: provider_module: <module 'faker.providers.credit_card' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/credit_card/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.currency
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.currency
FAKER: provider_module: <module 'faker.providers.currency' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/currency/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.date_time
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.date_time
FAKER: provider_module: <module 'faker.providers.date_time' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/date_time/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.en_US
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.en_US
FAKER: provider_module: <module 'faker.providers.en_US' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/en_US/__init__.pyc'>
```

**Comment by whit537** [comment]
Local working `envtest tests/py/test_stats.py`:

```
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.address
FAKER: using localized provider module
FAKER: available locales: ['cs_CZ', 'de_DE', 'el_GR', 'en', 'en_AU', 'en_CA', 'en_GB', 'en_US', 'es', 'es_ES', 'es_MX', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hi_IN', 'hr_HR', 'it_IT', 'ja_JP', 'ko_KR', 'ne_NP', 'nl_NL', 'no_NO', 'pl_PL', 'pt_BR', 'pt_PT', 'sk_SK', 'sl_SI', 'sv_SE', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.address.en_US
FAKER: provider_module: <module 'faker.providers.address.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/address/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.barcode
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.barcode
FAKER: provider_module: <module 'faker.providers.barcode' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/barcode/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.color
FAKER: using localized provider module
FAKER: available locales: ['en_US']
FAKER: path: faker.providers.color.en_US
FAKER: provider_module: <module 'faker.providers.color.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/color/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.company
FAKER: using localized provider module
FAKER: available locales: ['bg_BG', 'cs_CZ', 'de_DE', 'en_US', 'es_MX', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hr_HR', 'it_IT', 'ja_JP', 'ko_KR', 'no_NO', 'pt_BR', 'pt_PT', 'sk_SK', 'sl_SI', 'sv_SE', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.company.en_US
FAKER: provider_module: <module 'faker.providers.company.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/company/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.credit_card
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.credit_card
FAKER: provider_module: <module 'faker.providers.credit_card' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/credit_card/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.currency
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.currency
FAKER: provider_module: <module 'faker.providers.currency' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/currency/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.date_time
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.date_time
FAKER: provider_module: <module 'faker.providers.date_time' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/date_time/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.file
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.file
FAKER: provider_module: <module 'faker.providers.file' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/file/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.internet
FAKER: using localized provider module
FAKER: available locales: ['bg_BG', 'bs_BA', 'cs_CZ', 'de_AT', 'de_DE', 'el_GR', 'en_AU', 'en_US', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hr_HR', 'ja_JP', 'ko_KR', 'no_NO', 'pt_BR', 'pt_PT', 'sk_SK', 'sl_SI', 'sv_SE', 'zh_CN']
FAKER: path: faker.providers.internet.en_US
FAKER: provider_module: <module 'faker.providers.internet.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/internet/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.job
FAKER: using localized provider module
FAKER: available locales: ['en_US', 'fa_IR', 'fr_CH', 'fr_FR', 'hr_HR', 'pl_PL', 'ru_RU', 'uk_UA', 'zh_TW']
FAKER: path: faker.providers.job.en_US
FAKER: provider_module: <module 'faker.providers.job.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/job/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.lorem
FAKER: using localized provider module
FAKER: available locales: ['el_GR', 'la', 'ru_RU']
FAKER: path: faker.providers.lorem.la
FAKER: provider_module: <module 'faker.providers.lorem.la' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/lorem/la/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.misc
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.misc
FAKER: provider_module: <module 'faker.providers.misc' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/misc/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.person
FAKER: using localized provider module
FAKER: available locales: ['bg_BG', 'cs_CZ', 'de_AT', 'de_DE', 'dk_DK', 'el_GR', 'en', 'en_GB', 'en_US', 'es_ES', 'es_MX', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hi_IN', 'hr_HR', 'it_IT', 'ja_JP', 'ko_KR', 'lt_LT', 'lv_LV', 'ne_NP', 'nl_NL', 'no_NO', 'pl_PL', 'pt_BR', 'pt_PT', 'ru_RU', 'sl_SI', 'sv_SE', 'tr_TR', 'uk_UA', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.person.en_US
FAKER: provider_module: <module 'faker.providers.person.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/person/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.phone_number
FAKER: using localized provider module
FAKER: available locales: ['bg_BG', 'bs_BA', 'cs_CZ', 'de_DE', 'dk_DK', 'el_GR', 'en_AU', 'en_CA', 'en_GB', 'en_US', 'es_ES', 'es_MX', 'fa_IR', 'fi_FI', 'fr_CH', 'fr_FR', 'hi_IN', 'hr_HR', 'it_IT', 'ja_JP', 'ko_KR', 'lt_LT', 'lv_LV', 'ne_NP', 'nl_NL', 'no_NO', 'pl_PL', 'pt_BR', 'pt_PT', 'ru_RU', 'sk_SK', 'sl_SI', 'sv_SE', 'tr_TR', 'uk_UA', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.phone_number.en_US
FAKER: provider_module: <module 'faker.providers.phone_number.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/phone_number/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.profile
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.profile
FAKER: provider_module: <module 'faker.providers.profile' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/profile/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.python
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.python
FAKER: provider_module: <module 'faker.providers.python' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/python/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.ssn
FAKER: using localized provider module
FAKER: available locales: ['en_CA', 'en_US', 'fi_FI', 'fr_CH', 'hr_HR', 'it_IT', 'ko_KR', 'nl_NL', 'pt_BR', 'sv_SE', 'uk_UA', 'zh_CN', 'zh_TW']
FAKER: path: faker.providers.ssn.en_US
FAKER: provider_module: <module 'faker.providers.ssn.en_US' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/ssn/en_US/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.user_agent
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.user_agent
FAKER: provider_module: <module 'faker.providers.user_agent' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/user_agent/__init__.py'>
```

**Comment by whit537** [comment]
Here's where the breakdown occurs:

# Working Locally

```
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.date_time
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.date_time
FAKER: provider_module: <module 'faker.providers.date_time' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/date_time/__init__.py'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.file
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.file
FAKER: provider_module: <module 'faker.providers.file' from '/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/file/__init__.py'>
```

# Broken at Travis

```
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.date_time
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.date_time
FAKER: provider_module: <module 'faker.providers.date_time' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/date_time/__init__.pyc'>
FAKER: ------------------------------------------------------------------------------
FAKER: provider path: faker.providers.en_US
FAKER: not using localized provider module
FAKER: locale is not None; faker.providers.en_US
FAKER: provider_module: <module 'faker.providers.en_US' from '/home/travis/build/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers/en_US/__init__.pyc'>
```

**Comment by whit537** [comment]
With dffcac0ef72a476582af8b2eb1bc91efa585334f:

# Working Locally

`envtest tests/py/test_stats.py`

```
FAKER: ***************************************************************************
FAKER: providers: None
FAKER: PROVIDERS: ['faker.providers.address', 'faker.providers.barcode', 'faker.providers.color', 'faker.providers.company', 'faker.providers.credit_card', 'faker.providers.currency', 'faker.providers.date_time', 'faker.providers.file', 'faker.providers.internet', 'faker.providers.job', 'faker.providers.lorem', 'faker.providers.misc', 'faker.providers.person', 'faker.providers.phone_number', 'faker.providers.profile', 'faker.providers.python', 'faker.providers.ssn', 'faker.providers.user_agent']
FAKER: includes: []
```

# Broken at Travis

```
___________________ ERROR collecting tests/py/test_stats.py ____________________
tests/py/test_stats.py:8: in <module>
    from gratipay.utils import fake_data
gratipay/utils/fake_data.py:19: in <module>
    faker = Factory.create()
env/lib/python2.7/site-packages/faker/factory.py:45: in create
    prov_cls, lang_found = cls._get_provider_class(prov_name, locale)
env/lib/python2.7/site-packages/faker/factory.py:56: in _get_provider_class
    provider_class = cls._find_provider_class(provider, locale)
env/lib/python2.7/site-packages/faker/factory.py:104: in _find_provider_class
    return provider_module.Provider
E   AttributeError: 'module' object has no attribute 'Provider'
------------------------------- Captured stdout --------------------------------
FAKER: ***************************************************************************
FAKER: providers: None
FAKER: PROVIDERS: ['faker.providers.address', 'faker.providers.barcode', 'faker.providers.color', 'faker.providers.company', 'faker.providers.credit_card', 'faker.providers.currency', 'faker.providers.date_time', 'faker.providers.en_US', 'faker.providers.file', 'faker.providers.fr_FR', 'faker.providers.internet', 'faker.providers.it_IT', 'faker.providers.job', 'faker.providers.lorem', 'faker.providers.misc', 'faker.providers.person', 'faker.providers.phone_number', 'faker.providers.profile', 'faker.providers.pt_BR', 'faker.providers.python', 'faker.providers.ssn', 'faker.providers.user_agent']
FAKER: includes: []
```

**Comment by whit537** [comment]
The broken version at Travis includes `faker.providers.en_US` in `PROVIDERS`, whereas the local working version does not. Is this a caching issue after all? Is this a vestige from the previous version install?

https://docs.travis-ci.com/user/caching/#Clearing-Caches

**Comment by whit537** [comment]
Cache cleared and build restarted ... 🙈

**Comment by whit537** [comment]
Not a cache issue. Hmm ... 8faba0e1dde6bb3528f1617bd9c3d14342b06d2c to confirm https://github.com/joke2k/faker/issues/332#issuecomment-190325168 ...

**Comment by whit537** [comment]
Confirmed.

```
0.01s$ ls -FGl env/lib/python2.7/site-packages/faker/providers/
total 312
drwxrwxr-x 33 travis  4096 Dec  7 17:27 address/
-rw-rw-r--  1 travis  2455 Dec  5 00:08 address.py
-rw-rw-r--  1 travis  4644 Dec  5 00:09 address.pyc
drwxrwxr-x  3 travis  4096 Dec  7 17:27 barcode/
drwxrwxr-x  3 travis  4096 Dec  7 17:27 color/
drwxrwxr-x 23 travis  4096 Dec  7 17:27 company/
-rw-rw-r--  1 travis   509 Dec  5 00:08 company.py
-rw-rw-r--  1 travis  1278 Dec  5 00:09 company.pyc
drwxrwxr-x  3 travis  4096 Dec  7 17:27 credit_card/
-rw-rw-r--  1 travis  6733 Dec  5 00:08 credit_card.py
-rw-rw-r--  1 travis  7160 Dec  5 00:09 credit_card.pyc
drwxrwxr-x  3 travis  4096 Dec  7 17:27 currency/
drwxrwxr-x  3 travis  4096 Dec  7 17:27 date_time/
-rw-rw-r--  1 travis 32355 Dec  5 00:08 date_time.py
-rw-rw-r--  1 travis 30867 Dec  5 00:09 date_time.pyc
drwxrwxr-x  2 travis  4096 Dec  5 00:09 en_US/
drwxrwxr-x  3 travis  4096 Dec  7 17:27 file/
-rw-rw-r--  1 travis  8070 Dec  5 00:08 file.py
-rw-rw-r--  1 travis  4165 Dec  5 00:09 file.pyc
drwxrwxr-x  2 travis  4096 Dec  5 00:09 fr_FR/
-rw-rw-r--  1 travis  6330 Dec  7 17:27 __init__.py
-rw-rw-r--  1 travis  9647 Dec  7 17:27 __init__.pyc
drwxrwxr-x 24 travis  4096 Dec  7 17:27 internet/
-rw-rw-r--  1 travis  4044 Dec  5 00:08 internet.py
-rw-rw-r--  1 travis  8051 Dec  5 00:09 internet.pyc
drwxrwxr-x  2 travis  4096 Dec  5 00:09 it_IT/
drwxrwxr-x 11 travis  4096 Dec  7 17:27 job/
drwxrwxr-x  5 travis  4096 Dec  7 17:27 lorem/
-rw-rw-r--  1 travis  7250 Dec  5 00:08 lorem.py
-rw-rw-r--  1 travis  9957 Dec  5 00:09 lorem.pyc
drwxrwxr-x  3 travis  4096 Dec  7 17:27 misc/
-rw-rw-r--  1 travis  1859 Dec  5 00:08 miscelleneous.py
-rw-rw-r--  1 travis  3632 Dec  5 00:09 miscelleneous.pyc
drwxrwxr-x 37 travis  4096 Dec  7 17:27 person/
-rw-rw-r--  1 travis   568 Dec  5 00:08 person.py
-rw-rw-r--  1 travis  1498 Dec  5 00:09 person.pyc
drwxrwxr-x 39 travis  4096 Dec  7 17:27 phone_number/
-rw-rw-r--  1 travis   235 Dec  5 00:08 phone_number.py
-rw-rw-r--  1 travis   911 Dec  5 00:09 phone_number.pyc
drwxrwxr-x  3 travis  4096 Dec  7 17:27 profile/
drwxrwxr-x  2 travis  4096 Dec  5 00:09 pt_BR/
drwxrwxr-x  3 travis  4096 Dec  7 17:27 python/
-rw-rw-r--  1 travis  4355 Dec  5 00:08 python.py
-rw-rw-r--  1 travis  5768 Dec  5 00:09 python.pyc
drwxrwxr-x 15 travis  4096 Dec  7 17:27 ssn/
drwxrwxr-x  3 travis  4096 Dec  7 17:27 user_agent/
-rw-rw-r--  1 travis  5274 Dec  5 00:08 user_agent.py
-rw-rw-r--  1 travis  6778 Dec  5 00:09 user_agent.pyc
```

**Comment by whit537** [comment]
Local:

```
/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/faker/providers
total 8
drwxr-xr-x  21 whit537  staff   714 Dec  6 17:19 ./
drwxr-xr-x  14 whit537  staff   476 Dec  7 11:54 ../
-rw-r--r--   1 whit537  staff  6330 Sep 16 00:57 __init__.py
drwxr-xr-x  34 whit537  staff  1156 Dec  6 17:19 address/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 barcode/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 color/
drwxr-xr-x  24 whit537  staff   816 Dec  6 17:19 company/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 credit_card/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 currency/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 date_time/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 file/
drwxr-xr-x  25 whit537  staff   850 Dec  6 17:19 internet/
drwxr-xr-x  12 whit537  staff   408 Dec  6 17:19 job/
drwxr-xr-x   6 whit537  staff   204 Dec  6 17:19 lorem/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 misc/
drwxr-xr-x  38 whit537  staff  1292 Dec  6 17:19 person/
drwxr-xr-x  40 whit537  staff  1360 Dec  6 17:19 phone_number/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 profile/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 python/
drwxr-xr-x  16 whit537  staff   544 Dec  6 17:19 ssn/
drwxr-xr-x   4 whit537  staff   136 Dec  6 17:19 user_agent/
$
```

**Comment by whit537** [comment]
I note that there are also `address.py`, etc. files at Travis, though no `color.py` ... because we never import that?

**Comment by whit537** [comment]
Wait! 8311327a33b084f5b7469aa87ffae8b1057d1ff8 is [green](https://travis-ci.org/gratipay/gratipay.com/builds/182030240)!?

**Comment by whit537** [comment]
But 93b50991859fd46cca09d314f0c6000dce24c34b is [red](https://travis-ci.org/gratipay/gratipay.com/builds/182034731) again? 😖

**Comment by whit537** [comment]
Different red.

**Comment by whit537** [comment]
Green! Does caching predict the result?

**Comment by mattbk** [comment]
image

**Comment by whit537** [comment]
Ooooh! Green instead?

**Comment by whit537** [comment]
> If a branch does not have its own cache, Travis CI fetches the master branch cache.

**Comment by whit537** [comment]
I'm going to try removing all caches.

**Comment by whit537** [comment]
My hypothesis is that when we don't specify the cache directory for the branch, Travis is falling back to a spoiled cache from master or something like that.

**Comment by whit537** [comment]
Well, that should be green, but it won't give us info about the hypothesis that a fall-through is causing failures to persist. For that I think we would need to remove all caches after all.

**Comment by whit537** [comment]
> No caches have been created yet,
> read more on [how to setup caching with your build](http://docs.travis-ci.com/user/caching/).

**Comment by whit537** [comment]
Here we go!

**Comment by whit537** [comment]
💃

**Comment by whit537** [comment]
Debugging commits rebased off. Was b2a33f9caba39f41f24c1c0925265c4d0e05ade7.

**Comment by whit537** [comment]
Build restarted for 45cb68da482d4a44a3f6224074ed0aa35e4a44fa ...

File: requirements.txt (hunk 1 above)
Comment (whit537) [inline requirements.txt]: Same with the next two lines, actually.

**Comment by whit537** [comment]
Green! Ready to declare victory here, @mattbk?

**Comment by mattbk** [comment]
Nice work!

**Pull request merged**